profiler/src/ProfilerEngine/Datadog.Profiler.Native/LibddprofExporter.cpp

```diff
@@ -152,6 +153,12 @@ void LibddprofExporter::RegisterUpscaleProvider(IUpscaleProvider* provider)
     _upscaledProviders.push_back(provider);
 }
 
+void LibddprofExporter::RegisterProcessSamplesProvider(ISamplesProvider* provider)
+{
+    assert(provider != nullptr);
```

Comment:
Need to just return if provider == nullptr

Reply:
it's the same pattern as register the upscaling provider